client/state/jetpack-connect/selectors.js

```diff
@@ -52,7 +52,7 @@ const isCalypsoStartedConnection = function( state, siteSlug ) {
 	if ( ! siteSlug ) {
 		return false;
 	}
-	const site = siteSlug.replace( /.*?:\/\//g, '' );
+	const site = siteSlug.replace( /.*?:\/\//g, '' ).replace( /\//g, '::' );
```

Comment:
also here... if siteSlug is what it names implies, there shouldn't be any `/` on it already  .. we have a weird mix of urls and slugs all around here :/